zcash_client_sqlite/src/wallet/transparent.rs

```diff
@@ -479,11 +492,19 @@ pub(crate) fn put_received_transparent_utxo<P: consensus::Parameters>(
 }
 
 /// Returns the vector of [`TxId`]s for transactions for which spentness state is indeterminate.
-pub(crate) fn transaction_data_requests(
+pub(crate) fn transaction_data_requests<P: consensus::Parameters>(
     conn: &rusqlite::Connection,
+    params: &P,
 ) -> Result<Vec<TransactionDataRequest>, SqliteClientError> {
     let mut tx_retrieval_stmt =
         conn.prepare_cached("SELECT txid, query_type FROM tx_retrieval_queue")?;
+    let mut address_request_stmt = conn.prepare_cached(
+        "SELECT ssq.address, t.target_height
+         FROM transparent_spend_search_queue ssq
+         JOIN transactions t ON t.id_tx = ssq.transaction_id
+         WHERE t.mined_height IS NULL
+         AND t.target_height IS NOT NULL",
+    )?;
```

Comment:
I think that the `WHERE` conditions here are incorrect. In particular, `t.target_height IS NOT NULL` is going to prevent us looking up the second transaction in a ZIP 320 pair, because we don't set `target_height` for the first transaction when we detect it via `store_decrypted_transaction`. Similarly, on recovery from seed, the first ZIP 320 transaction will have non-null `mined_height` which would _also_ currently prevent us from looking up the second transaction.

Reply:
The second transaction in a ZIP 320 pair, when created by this wallet, _will_ correctly have `target_height` set because it will ordinarily be populated by `store_transaction_to_be_sent`. However, you're correct that this will fail to achieve the desired goal for transactions discovered via the wallet recovery process.